### PR TITLE
kep-1965: update doc references for APIServerIdentity

### PR DESCRIPTION
**Pull request by andrewsykim**
Signed-off-by: Andrew Sy Kim <andrewsy@google.com>

Update feature gate reference for APIServerIdentity. Currently on the fence about whether we want more comprehensive documentation for this feature. In it's current state it is mainly going to be used for internal optimizations (i.e. StorageVersion API) and likely won't see too much use from users.

### CODE REVIEW

**Comment by sftim** [comment]
I recommend adding a short, simple page inside [https://kubernetes.io/docs/concepts/overview/working-with-objects/](https://kubernetes.io/docs/concepts/architecture/): `https://kubernetes.io/docs/concepts/architecture/leases/`

In that page, outline what a Lease is and where Kubernetes uses them.

**Comment by andrewsykim** [comment]
> I recommend adding a short, simple page inside [https://kubernetes.io/docs/concepts/overview/working-with-objects/](https://kubernetes.io/docs/concepts/architecture/): https://kubernetes.io/docs/concepts/architecture/leases/

> In that page, outline what a Lease is and where Kubernetes uses them.

I agree this would be useful, but admittedly also going outside the scope of this KEP to write general documentation about Leases. I will try to find the time to work on this but no promises :)

**Comment by sftim** [comment]
Second-mover disadvantage: the SIG that makes the general API gets to write up specific, whereas making good use of the existing API means the new documentation only makes sense following a tidying and reorganization.

For GA, we'll want the docs about API server identity leases to have a home _somewhere_ (and not just in the list of former feature gates). Optional but also recommened: revise pages within https://kubernetes.io/docs/tasks/debug/debug-cluster/

**Comment by andrewsykim** [comment]
I think general documentation about Leases and their uses (kube-apiserver identtiy being one of many) would be good to have, and I am happy to try to work on it during regular development cycle. I just don't have time to work on this right now.

File: content/en/docs/reference/command-line-tools-reference/feature-gates.md [hunk 1]
```diff
@@ -606,7 +607,7 @@ Each feature gate is designed for enabling/disabling a specific feature:
   filesystem walk for better performance and accuracy.
 - `LogarithmicScaleDown`: Enable semi-random selection of pods to evict on controller scaledown
   based on logarithmic bucketing of pod timestamps.
-- `MatchLabelKeysInPodTopologySpread`: Enable the `matchLabelKeys` field for 
+- `MatchLabelKeysInPodTopologySpread`: Enable the `matchLabelKeys` field for
```
Comment (andrewsykim) [inline content/en/docs/reference/command-line-tools-reference/feature-gates.md:610]:
```suggestion
- `MatchLabelKeysInPodTopologySpread`: Enable the `matchLabelKeys` field for 
```

**Comment by krol3** [comment]
Hi @andrewsykim! This PR needs a doc review by **Mon Nov 28th** to get this into the release. Please reach out to required SIGs to get their review. Thank you!

File: content/en/docs/reference/command-line-tools-reference/feature-gates.md [hunk 2]
```diff
@@ -62,7 +62,8 @@ For a reference to old feature gates that are removed, please refer to
 | `APIPriorityAndFairness` | `true` | Beta | 1.20 | |
 | `APIResponseCompression` | `false` | Alpha | 1.7 | 1.15 |
 | `APIResponseCompression` | `true` | Beta | 1.16 | |
-| `APIServerIdentity` | `false` | Alpha | 1.20 | |
+| `APIServerIdentity` | `false` | Alpha | 1.20 | 1.26 |
```
Comment (sftim) [inline content/en/docs/reference/command-line-tools-reference/feature-gates.md:65]:
```suggestion
| `APIServerIdentity` | `false` | Alpha | 1.20 | 1.25 |
```

**Review by sftim — COMMENTED** [review]
As well as documenting the feature gate, please document how Kubernetes behaves when this is enabled. A few lines may be enough; zero documentation is **not** sufficient.

See https://github.com/kubernetes/website/pull/37921#issuecomment-1316297747

**Comment by leonardpahlke** [comment]
x posting my [slack comment](https://kubernetes.slack.com/archives/C0156PJ62RE/p1669634876905699?thread_ts=1669632044.306809&cid=C0156PJ62RE)

I agree, the PR does not add sufficient documentation just yet.

About the PR [comment](https://github.com/kubernetes/website/pull/37921#issuecomment-1316327302):
> (@andrewsykim) I agree this would be useful, but admittedly also going outside the scope of this KEP to write general documentation about Leases.

Seems to me like a bit like an ownership issue, who “bootstraps general docs”. Perhaps we have missed to introduce the documentation page a while back?
I am +1 adding the documentation & suggestions @sftim mentioned now.

**Comment by sftim** [comment]
@andrewsykim if you write the text for the `APIServerIdentity` part, SIG Docs can write up any generic text that the new bit needs in order to have context.

**Comment by andrewsykim** [comment]
I added some general docs for Leases, PTAL.

Docs definitely  can use more details, but trying to be mindful of the docs deadline today. Definitely needs to be a follow-up to add more content in this area.

File: content/en/docs/concepts/architecture/leases.md [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+See [Node Lease objects](/docs/concepts/architecture/nodes/#heartbeats) for more details.
+
+## Leader Election
+
+Leases are also used in Kubernetes to ensure only one instance of a component is running at any given time.
+This is used by control plane components like `kube-controller-manager` and `kube-scheduler` in
+HA configurations, where only one instance of the component should be actively running while the other
+instances are on stand-by.
+
+## kube-apiserver identity
```
Comment (sftim) [inline content/en/docs/concepts/architecture/leases.md:32]:
```suggestion
## API Server identity
```

**Review by sftim — COMMENTED** [review]
LGTM for docs

I don't think this needs tech review; I can see that these pages look right enough.

/lgtm
/approve

**Comment by sftim** [comment]
After the release, we can track an issue to mention that workloads / out-of-tree things are also welcome to use Leases (especially controllers and operators).

File: content/en/docs/concepts/architecture/leases.md [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+
+{{< feature-state for_k8s_version="v1.26" state="beta" >}}
+
+Starting in Kubernetes v1.26, each `kube-apiserver` uses the Lease API to publish its identity to the
```
Comment (enj) [inline content/en/docs/concepts/architecture/leases.md:36]: Should we provide more details on the specifics here?  Like if I want to build an external integration that uses this information, how would I know what to look for?

Comment (andrewsykim) [reply content/en/docs/concepts/architecture/leases.md:36]: Added more details in https://github.com/kubernetes/website/pull/38196

**Pull request merged**